### PR TITLE
Moving code coverage from codecov package to CI

### PR DESCRIPTION
**Pull request by mamankhan99**
Codecov NodeJS Uploader [package](https://www.npmjs.com/package/codecov) is depreciated since version 3.8.3

This PR move the coverage to CI according to the new guidelines and [depreciation](https://about.codecov.io/blog/codecov-uploader-deprecation-plan/) plan of codecov

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (arbrandes) [inline .github/workflows/ci.yml:40]: Missing newline at EOF.

**Review by arbrandes — CHANGES_REQUESTED** [review]
Looks good, modulo a minor nitpick.

**Comment by mamankhan99** [comment]
@arbrandes waiting for approval to merge this PR.

**Review by arbrandes — APPROVED** [review]
Approved!  Thanks!

**Pull request merged**
